Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```diff
@@ -5,43 +5,156 @@
+		
+		public bool DeleteUploadedRequestsFromRelativityOM(IObjectManager proxy, int workspaceID, List<WikitivityUploadsAgent.SingleRequestObject> listOfCompletedUploads)
 		{
-			proxy.APIOptions.WorkspaceID = workspaceID;
-
 			bool success = false;
 			try
 			{
-				List<Document> DocsToDelete = new List<Document>();
+				List<RelativityObjectRef> listofObjectstoDelete = new List<RelativityObjectRef>();
 				foreach (var singleDoc in listOfCompletedUploads)
 				{
-					DocsToDelete.Add(new Document(singleDoc.ArtifactID));
+					listofObjectstoDelete.Add(new RelativityObjectRef() { ArtifactID = singleDoc.ArtifactID });
 				}
 
-				var DeleteResultSet = proxy.Repositories.Document.Delete(DocsToDelete);
-				if (DeleteResultSet.Success)
+				var deleteRequest = new MassDeleteByObjectIdentifiersRequest();
+				// Represents a list of RelativityObjects to be deleted.
+				deleteRequest.Objects = listofObjectstoDelete;
+				Task.Run(async () =>
+					{
+						await proxy.DeleteAsync(workspaceID, deleteRequest);
+					}).Wait();
+				return success = true;
+			}
+			catch (Exception ex)
+			{
+				throw new Exception("An error an occurred when deleting completed requests", ex);
+			}
+		}
+
+		public static async Task<List<int>> FindWikitivityCasesOM(IObjectManager proxy, int workspaceID)
+		{
+			QueryResultSlim test = new QueryResultSlim();
+			List<int> CasesWithWikitivity = new List<int>();
+
+			QueryRequest omGetInitialJobQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request Url' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, 
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
 				{
-					success = true;
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"}
+
 				}
+			};
+			try
+			{
+				test = await proxy.QuerySlimAsync(workspaceID, omGetInitialJobQuery, 1, 10000);
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine(e);
+				throw;
+			}
+
+			foreach (var singleWorkspace in test.Objects)
+			{
+				CasesWithWikitivity.Add(singleWorkspace.ArtifactID);
+			}
+			return CasesWithWikitivity;
+		}
+		
+		public static async Task<QueryResultSlim> FindAllWikiCases(IObjectManager proxy, int workspaceID)
+		{
+
+
+			QueryRequest omGetInitialJobQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request Url' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, //name of relational field to expand query results to related objects
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
+				{
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"}
+				}
+			};

```

Comment:
This looks like the same query.  You should can make a helper function for building your queries so you would only need to make changes in one spot.